brush-core/src/builtins/factory.rs

```diff
@@ -283,6 +283,7 @@ pub(crate) fn get_default_builtins(
         m.insert("enable".into(), builtin::<enable::EnableCommand>());
         m.insert("let".into(), builtin::<let_::LetCommand>());
         m.insert("mapfile".into(), builtin::<mapfile::MapFileCommand>());
+        m.insert("readarray".into(), builtin::<mapfile::MapFileCommand>());
```

Comment:
You'll need to remove lines 321-322. That would also explain the test failures--the stub registered below is overriding the registration here.